core/src/main/java/org/infinispan/interceptors/EntryWrappingInterceptor.java

```diff
@@ -200,8 +200,8 @@ public final Object visitPutKeyValueCommand(InvocationContext ctx, PutKeyValueCo
 
    private void wrapEntryForPutIfNeeded(InvocationContext ctx, AbstractDataWriteCommand command) throws Throwable {
       if (shouldWrap(command.getKey(), ctx, command)) {
-         entryFactory.wrapEntryForPut(ctx, command.getKey(), null, !command.isConditional(), command,
-            command.hasFlag(Flag.IGNORE_RETURN_VALUES) && !command.isConditional());
+         entryFactory.wrapEntryForPut(ctx, command.getKey(), null, false, command,
```

Comment:
some invocations could be optimized. we can skip the read when ignore return values flags is used for example. Although, it may need to check if the key exists to trigger the correct notification.